### DISCUSSION TITLE
Need for variables for template repositories.

### DISCUSSION THREAD

**Discussion by itsmeadarsh2008** [community/community#127141] [Repositories]
### Select Topic Area

General

### Body

Whenever a coder makes a repo from a template repo, it will ask questions for substituting variables in the repo before making an initial commit. This will make it easier to make projects with custom data.

**Comment by Ochieng-001** [comment]
Very true because template repositories with variable substitution let coders customize projects easily. They answer prompts for variables during creation, replacing placeholders in the template with their specific values. This saves time and ensures consistency for projects based on the same starting point.

**Comment by patrick-knight** [comment]
Thanks for the suggestion @itsmeadarsh2008. Can you give me a few examples of variables that you are thinking about that would be helpful? 

Transparently, my concern with patterns like this is where do you draw then line on number of supported variables? One or two requirements before cloning a template isn't bad, but is 10, 20, 100 too many?

**Reply by itsmeadarsh2008** [reply]
This is like an issue template-like feature, what to ask for the variable's value is defined in a file. So, the number of variables depends on how much the contributor defines, regarding setting a bar for variables. We can do a simple thing, we will set it to 45-75 and to increase the bar for a repository, the repository holder has to request for increase in the bar. I think, no one will need more than 45 only if they aren't an organization.

**Reply by mo10** [reply]
Solution of Gitea: https://docs.gitea.com/usage/template-repositories

**Reply by itsmeadarsh2008** [reply]
@mo10 It's crazy that Gitea supports this but not GitHub
